delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3FileOperations.scala

```diff
@@ -63,6 +74,34 @@ object S3FileOperations {
 
     override def save(storage: S3Storage, filename: String, entity: BodyPartEntity): IO[FileStorageMetadata] =
       saveFile.apply(storage, filename, entity)
+
+    override def register(bucket: String, path: Uri.Path): IO[S3FileMetadata] = {
+      for {
+        _           <- log.info(s"Fetching attributes for S3 file. Bucket $bucket at path $path")
+        resp        <- client.headObject(bucket, path.toString())
```

Comment:
Why not using GetObjectAttributes:
https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObjectAttributes.html

Reply:
Hum, I just thought that head did not return the required info but if it does, I am fine with it